node/routes/workspaceSearchParams.ts

```diff
@@ -0,0 +1,35 @@
+const BUCKET = 'workspace-search-params'
+const MASTER = 'master'
+
+function disableCache(ctx: Context) {
+    ctx.set('Cache-control', 'no-cache,no-store')
+}
+
+export async function getWorkspaceSearchParamsFromStorage(ctx: Context) {   
+    const masterParams = await ctx.clients.vbase.getJSON<object>(BUCKET, MASTER, true)
+    if (ctx.vtex.workspace == MASTER) return masterParams
+    const filePath = ctx.vtex.workspace
+    const params = await ctx.clients.vbase.getJSON<object>(BUCKET, filePath, true)
+    // Logic: workspaces inherit params from master, and can override as desired
+    return {
+        ...masterParams,
+        ...params
+    }
+}
+
+export async function getWorkspaceSearchParams(ctx: Context, next: () => Promise<object>) {
+    const data = await getWorkspaceSearchParamsFromStorage(ctx)
+    disableCache(ctx)
```

Comment:
sorry, why do we need to disable the cache? I remember that you mentioned the reason, but I can't remember

Reply:
This was when from when I was using public routes, so that CDNs don't cache the result. Now that it is under private routes, maybe no longer necessary, or is there any caching in internal routes? Either way, maybe no harm in leaving it. It is for internal use to get the current params.

Reply:
Agree, since it is only an intern configuration I don't see any problem. I was just curious about it. Nice! we are done here 😉

Reply:
Agree, since it is only an intern configuration I don't see any problem. I was just curious about it. Nice! we are done here 😉